<a href="https://colab.research.google.com/github/etrompetam/Codigo/blob/main/04_08_03_ejemplo_Main.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Relacion entre los esferzos principales obtenidos en el analisis bidimencional y tridimencional**

se importan las librerias necesarias:

In [16]:
import numpy as np
# np.set_printoption(precision)

In [17]:
# Se definen las variables
sx  =  -1
sy  =   2
txy =  -3
nu  =   0.30

# Matriz de esfuerzos.
sigma = np.array([[ sx,  txy,              0],
                  [txy,   sy,              0],
                  [  0,    0,   nu*(sx + sy)]])

In [18]:
sigma

array([[-1. , -3. ,  0. ],
       [-3. ,  2. ,  0. ],
       [ 0. ,  0. ,  0.3]])

In [19]:
# Calculo de valores y vectores propios
valp, vecp = np.linalg.eigh(sigma)

In [20]:
# Los valores propios salen ordenados de menor a mayor [s3, s2, s1]
valp

array([-2.85410197,  0.3       ,  3.85410197])

In [25]:
# Los vectores propios correspondientes [n3, n2, n1]
vecp

array([[-0.85065081,  0.        , -0.52573111],
       [-0.52573111,  0.        ,  0.85065081],
       [-0.        ,  1.        ,  0.        ]])

verificamos que los vectores principales formen un sistema coordenado de la mano derecha, a partir del producto vectorial $(\mathbf{ \hat{n}_3 = \hat{n}_1 \times \hat{n}_2 }):$

In [27]:
# Extraigo las direcciones principales de calculo anterior.
n3, n2, n1 = vecp.T

# Calculo n3 = n1 x n2
n3_calculado = np.cross(n1, n2)
n3_calculado

array([ 0.85065081,  0.52573111, -0.        ])

Fijese que obtenemos el vector $\mathbf{ \hat{n}_3}$ con sentido contrario, esto se puede solucionar considerando en sentido contrario a $\mathbf{ \hat{n}_2}$

In [28]:
# Calculo n3 = n1 x -n2
n3_calculado = np.cross(n1, -n2)
n3_calculado

array([-0.85065081, -0.52573111,  0.        ])

con lo cual cumplimos la regla de la mano derecha, al obtener el sentido requerido para $\hat{n}_3$.

## Conclusión del main.pdf

Observe la discrepancia de signo entre los vectores propios $[− 0.8507, − 0.5257, 0]^\intercal$ y $[0.8507, 0.5257, 0]^\intercal$; esta diferencia no nos debe importar, ya que lo único relevante en el cálculo de los vectores propios es la dirección del vector mismo, no su sentido. De esta forma, si hubiéramos escogido el vector $\hat{n}_3$ como $[− 0.8507, − 0.5257, 0]^\intercal$, el sentido del vector $\hat{n}_2$ igualmente hubiera cambiado de modo que $\hat{n}_2$ hubiera sido $[0, 0, − 1]^\intercal$ con el fin de cumplir la regla de la mano derecha en la base formada por las direcciones principales.

Tenemos pues que dos de los valores propios corresponden a $(\sigma_1)_{xy}$  y $(\sigma_2)_{xy}$, mientras que el otro corresponde a $0$. Los vectores propios asociados son los mismos calculados en el caso bidimensional para los dos primeros, agregando un $0$ en la componente $z$, mientras que el vector propio asociado a 0 es [0,0, ± 1]T , siendo necesario ajustar su sentido (ver Figura 4.18).

Al igual que en el caso de deformación plana, debemos determinar la magnitud y los planos sobre los que actúa el esfuerzo cortante máximo.

En este ejemplo en particular,  estos son exactamente los mismos que los calculados en el ejemplo de deformación plana. Sin embargo, estos pueden variar. Supongamos que ambos $\sigma_1$ y $\sigma_2$ son cantidades positivas, entonces en este caso, tendremos un $\sigma_3 = 0$. Observe, por lo tanto, que en este caso cambiarían los valores de $\tau_{\max}$ y los vectores $\frac{ \hat{n}_1 - \hat{n}_3}{||\hat{n}_1 - \hat{n}_3||}$ y $\frac{ \hat{n}_1 + \hat{n}_3}{||\hat{n}_1 + \hat{n}_3||}$ ya no estarían sobre el plano $xy$.

En resumen, para el aso de tensión plana, se tiene que:

$$
\begin{align*}

\sigma_1    &= \max\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) \\
\sigma_2    &= mediana\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) \\
\sigma_3    &= \min\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, 0 \right) \\
\tau_{\max} &= \max\left( \frac{|(\sigma_1)_{xy}| }{2},  \frac{|(\sigma_2)_{xy}| }{2},  \frac{ |(\sigma_1)_{xy} - (\sigma_2)_{xy}| }{2}   \right)

\end{align*}
$$